services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -597,7 +598,24 @@ export class WorkflowService {
       reason?: string;
     },
   ) {
-    const runtimeData = await this.workflowRuntimeDataRepository.findById(workflowId);
+    const workflow = await this.workflowRuntimeDataRepository.findById(workflowId);
```

Comment:
Added validation step, handles document.type unknown, and does not validate when the decision is reject/revision.